File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int i = 0;
        int j = people.length - 1;
        int answer = 0;

        Arrays.sort(people);

        while (i <= j) {
            answer++;

            if (people[i] + people[j] <= limit) {
                i++;
            }

            j--;
        }

        return answer;
    }
}

File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    i, j, answer = 0, len(people) - 1, 0
    
    while i <= j:
        answer += 1
        
        if people[i] + people[j] <= limit:
            i += 1
            
        j -= 1
        
    return answer